microsoft-edge/devtools-guide-chromium/speed/get-started.md

```diff
@@ -22,440 +22,480 @@ ms.date: 05/04/2021
+The webpage uses the remaining CSS files so you can't remove them. However, it's possible to split them in two groups:
 
-   ![The Network panel shows that the requests have been blocked](../media/speed-glitch-tony-remix-updated-network-reloaded-drawer-request-blocking-added.msft.png)
+* Critical CSS code that needs to block the rendering of the webpage because it visually impacts the style and layout of the part of the webpage that users see when it loads.
 
-1. Click **Remove all patterns** (![Remove all patterns](../media/remove-icon.msft.png)) to delete the `/libs/*` blocking pattern.
+    For example, the title on the webpage uses the `header h1` CSS rule in the `/travel-site/assets/base.css` file.
 
-In general, the **Request blocking** tool is useful for simulating how your page behaves when any given resource isn't available.
+* Non-critical CSS code used to render parts of the page that aren't visible when the page loads.
 
-Now, remove the references to these files from the code and audit the page again:
+    For example, the `/travel-site/assets/desktop.css` file is only needed when the viewport is larger than `665px`.
 
-1. Back in the editor tab, open `template.html`.
+To automatically split your CSS code this way, you can use the _Critical_ tool. To learn more, see the [Critical project repository](https://github.com/addyosmani/critical) at GitHub.
 
-1. Delete `<script src="/libs/lodash.js">` and `<script src="/libs/jquery.js"></script>`.
+You can then load your non-critical CSS code in a way to doesn't block the initial render of the page. To learn more, see [Defer non-critical CSS](https://web.dev/defer-non-critical-css/) at web.dev.
 
-1. Wait for the site to re-build and re-deploy.
+It's also good to minify your CSS code and remove unneeded whitespace characters and comments. To learn more, see [Minify CSS](https://web.dev/minify-css/) at web.dev.
 
-1. Audit the page again from the **Audits** tool.  Your overall score should have improved again.
+#### Reduce layout shift by setting explicit width and height on images
 
-   ![An Audits report after removing the render-blocking resources](../media/speed-glitch-tony-remix-updated-2-audits-performance.msft.png)
+In your **Lighthouse** report, the **Diagnostics** section suggests defining explicit `width` and `height` for image elements. Click on **Image elements do not have explicit width and height**  to display more information:
 
-#### Optimizing the Critical Rendering Path in the real-world
+![The Lighthouse diagnostics section about setting explicit width and height to images](./images/explicit-image-size.png)
 
-The **Critical Rendering Path** refers to the code that you need to load a page.  In general, speed up page load by only shipping critical code during the page load, and then lazy-loading everything else.
+The report says that the image at the top of the webpage doesn't have an explicit `width` and `height`, which can cause layout shifts.
```

Comment:
```suggestion
The report says that the image at the top of the webpage doesn't have an explicit `width` and `height` setting, which can cause layout shifts.
```
I'd want to study context more, to double check if this improves

Reply:
I think we should use "attributes" instead here, as this is actually what's needed.